comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,176 @@ public static bool IsDirectory(string path)
 		}
 
 		// Get the path and size (in bytes) of the smallest file below the given directory
 		public static Tuple<string, long> GetSmallestFile(string directory)
 		{
-			throw new NotImplementedException();
+            long Smallest_File_Size = 0;
+            string Saved_Directory = "" ;
+
+            string[] files = Directory.GetFileSystemEntries(directory);
+
+            foreach (string element in files) // check's each file against recorded smallest and reasigns variables if smaller file is found. 
+            {
+                if (IsDirectory(element) == false)
+                {
+                    long File_Size = GetFileSize(element);
+
+                    if (Smallest_File_Size > File_Size || Smallest_File_Size == 0) // ensuring to eliminate the starting variables "" and 0
+                    {
+                        Smallest_File_Size = File_Size;
+                        Saved_Directory = element;
+                    }
+                }
+                else {
+                    Tuple <string, long>Smallest_In_Directory = GetSmallestFile(element);
+
+                    if (Smallest_File_Size > Smallest_In_Directory.Item2 || Smallest_File_Size == 0) {
+                        Smallest_File_Size = Smallest_In_Directory.Item2;
+                        Saved_Directory = Smallest_In_Directory.Item1;
+                    }
+                }
+            }
+
+            return (new Tuple<string, long>(Saved_Directory, Smallest_File_Size));                                  
 		}
 
 		// Get the path and size (in bytes) of the largest file below the given directory
 		public static Tuple<string, long> GetLargestFile(string directory)
 		{
-			throw new NotImplementedException();
+            long Largest_File_Size = 0;
+            string Saved_Directory = "";
+
+            string[] files = Directory.GetFileSystemEntries(directory);
+
+            foreach (string element in files)  // check's each file against recorded largest and reasigns variables if larger file is found.
+            {
+                if (IsDirectory(element) == false)
+                {
+                    long File_Size = GetFileSize(element);
+
+                    if (Largest_File_Size < File_Size )
+                    {
+                        Largest_File_Size = File_Size;
+                        Saved_Directory = element;
+                    }
+                }
+                else
+                {
+                    Tuple<string, long> Largest_In_Directory = GetLargestFile(element);
+
+                    if (Largest_File_Size < Largest_In_Directory.Item2)
+                    {
+                        Largest_File_Size = Largest_In_Directory.Item2;
+                        Saved_Directory = Largest_In_Directory.Item1;
+                    }
+                }
+            }
+
+            return (new Tuple<string, long>(Saved_Directory, Largest_File_Size));
+
+            throw new NotImplementedException();
```

Comment:
It's best to remove the `NotImplementedException` line when you're done -- it doesn't really make a difference here since it comes after the `return` hence is unreachable, but it is still a good idea.